sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -0,0 +1,239 @@
+   */
+  kDifRstmgrSoftwareReset = 0,
+  kDifRstmgrPeripheralResetHold,    /**< Hold peripheral in reset. */
+  kDifRstmgrPeripheralResetRelease, /**< Release peripheral from reset. */
+} dif_rstmgr_software_reset_t;
+
+/**
+ * Reset Manager bitfield encoding the reset causes information.
+ */
+typedef uint32_t dif_rstmgr_causes_bitfield_t;
+
+/**
+ * Reset Manager all possible reset causes enumeration.
+ *
+ * Invariants are used to extract information encoded in
+ * `dif_rstmgr_causes_bitfield_t`, which means that the values must correspond
+ * to the individual bits (0x1, 0x2, 0x4, ..., 0x80000000).
+ *
+ * Note: these must match the autogenerated register definitions.
+ *
+ * TODO: no documentation currently available, would be useful
+ *       for writing a more detailed description for the individual causes.
+ */
+typedef enum dif_rstmgr_cause {
+  kDifRstmgrCausePowerUp = 0x1,
+  kDifRstmgrCauseLowPowerExit = 0x2,
+  kDifRstmgrCauseWatchdog = 0x4,
+  kDifRstmgrCauseSecurityEscalation = 0x8,
+  kDifRstmgrCauseNonDebugModuleRequest = 0x10,
+} dif_rstmgr_cause_t;
+
+/**
+ * Reset Manager software reset available peripherals.
+ */
+typedef enum dif_rstmgr_peripheral {
+  kDifRstmgrPeripheralSpi = 0,
+  kDifRstmgrPeripheralUsb,
+  kDifRstmgrPeripheralLast = kDifRstmgrPeripheralUsb,
+} dif_rstmgr_peripheral_t;
+
+/**
+ * The result of a Reset Manager operation.
+ */
+typedef enum dif_rstmgr_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifRstmgrOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifRstmgrError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifRstmgrBadArg = 2,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifRstmgrLocked = 3,
+} dif_rstmgr_result_t;
+
+/**
+ * Hardware instantiation parameters for rstmgr.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_rstmgr_param {
+  /**
+   * The base address for the rstmgr hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_rstmgr_param_t;
+
+/**
+ * A handle to rstmgr.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_rstmgr { dif_rstmgr_param_t params; } dif_rstmgr_t;
+
+/**
+ * Creates a new handle for Reset Manager.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param handle Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_init(dif_rstmgr_param_t params,
+                                    dif_rstmgr_t *handle);
+
+/**
+ * Resets the Reset Manager registers to sane defaults.
```

Comment:
Yeah, I was struggling to describe it. Also, I think this function needs a check for whether these are locked, otherwise they won't be able to be de-asserted. Maybe this function should be completely renamed.